component/espejote-templates/netpol-sync.jsonnet

```diff
@@ -1,106 +1,170 @@
-  esp.markForDelete({
-    apiVersion: resources.allowFromSameNamespace.apiVersion,
-    kind: resources.allowFromSameNamespace.kind,
-    metadata: {
-      name: resources.allowFromSameNamespace.metadata.name,
-      namespace: namespace.metadata.name,
-    },
-  }),
-];
+// Extract the desired policy sets from the given namespace object,
+// Returns an empty array if the namespace is in the ignoredNamespaces list or the no-defaults label is set.
+// If no policy sets are set by labels, the default policy sets are returned and the legacy base-defaults label is respected.
+// If any policy set labels are set, those are returned plus the 'base' policy set.
+// It is valid to set the base policy set explicitly via labels.
+//   ignoredNamespaces / no-defaults label -> []
+//   no policy set labels -> [ 'base', 'default' ] or [ 'base' ] if base-defaults label is set
+//   policy set labels -> [ 'base', <policy sets from labels> ]
+//   policy set "base" set -> [ 'base' ]
+local desiredPolicySets(namespace) =
+  local objHasLabel(obj, label) =
+    std.objectHas(std.get(obj.metadata, 'labels', {}), label);
 
-local purgeDefaultPolicies(namespace) =
-  purgeNonBasePolicies(namespace) + [
-    esp.markForDelete({
-      apiVersion: resources.allowFromOtherNamespaces.apiVersion,
-      kind: resources.allowFromOtherNamespaces.kind,
-      metadata: {
-        name: resources.allowFromOtherNamespaces.metadata.name,
-        namespace: namespace.metadata.name,
-      },
-    }),
-    esp.markForDelete({
-      apiVersion: resources.allowFromClusterNodes.apiVersion,
-      kind: resources.allowFromClusterNodes.kind,
-      metadata: {
-        name: resources.allowFromClusterNodes.metadata.name,
-        namespace: namespace.metadata.name,
-      },
-    }),
-  ];
+  local defaultPolicySets =
+    if objHasLabel(namespace, config.namespaceLabels.baseDefaults) then
+      [ 'base' ]
+    else
+      [ 'base', 'default' ];
 
-local applyBasePolicies(namespace) = [
-  resources.allowFromOtherWithSpec {
-    metadata+: {
-      namespace: namespace.metadata.name,
-    },
-  },
-] + if config.networkPlugin == 'cilium' then [
-  resources.allowFromNodesWithSpec {
+  // Policy sets based on labels starting with params.labels.policySetPrefix.
+  //   labels:
+  //     set.example.io/airlock: ""
+  //     set.example.io/myapp: ""
+  // would return the policy sets `["airlock", "myapp"]`.
+  // The configured prefix is suffixed with a '/' if it does not already end with one.
+  local policySetsFromLabel =
+    local prefix = if std.endsWith(config.namespaceLabels.policySetPrefix, '/') then
+      config.namespaceLabels.policySetPrefix
+    else
+      config.namespaceLabels.policySetPrefix + '/';
+
+    [
+      lbl[std.length(prefix):]
+      for lbl in std.objectFields(std.get(namespace.metadata, 'labels', {}))
+      if std.startsWith(lbl, prefix) && (namespace.metadata.labels[lbl] == 'true' || namespace.metadata.labels[lbl] == '')
```

Comment:
Is the behavior of the set label values documented anywhere?

Reply:
It is now 😊